docs/architecture/application/graphql.md

```diff
@@ -37,3 +38,204 @@ app = OrchestratorCore(base_settings=AppSettings())
 # register SUBSCRIPTION_MODEL_REGISTRY
 app.register_graphql(query=NewQuery)
 ```
+
+## Domain Models Auto Registration for GraphQL
+
+When using the register_graphql function, all products in the SUBSCRIPTION_MODEL_REGISTRY will be automatically converted into GraphQL types.
+The registration process iterates through the list, starting from the deepest product block and working its way back up to the product level.
+
+However, there is a potential issue when dealing with a ProductBlock that references itself, as it can lead to an infinite loop.
+To handle this situation, you must manually create the GraphQL type for that ProductBlock and add it to the GRAPHQL_MODELS list.
+
+Here's an example of how to do it:
+
+```python
+import strawberry
+from typing import Annotated
+from app.product_blocks import ProductBlock
+from orchestrator.graphql import GRAPHQL_MODELS
+
+
+# It is necessary to use pydantic type, so that other product blocks can recognize it when typing to GraphQL.
+@strawberry.experimental.pydantic.type(model=ProductBlock)
+class ProductBlockGraphql:
+    name: strawberry.auto
+    self_refrence_block: Annotated[
+        "ProductBlockGraphql", strawberry.lazy(".product_block_file")
+    ] | None = None
+    ...
+
+
+# Add the ProductBlockGraphql type to GRAPHQL_MODELS, which is used in auto-registration for already created product blocks.
+GRAPHQL_MODELS.update({"ProductBlockGraphql": ProductBlockGraphql})
+```
+
+By following this example, you can effectively create the necessary GraphQL type for ProductBlock and ensure proper registration with register_graphql. This will help you avoid any infinite loop scenarios and enable smooth integration of domain models with GraphQL.
```

Comment:
```suggestion
By following this example, you can effectively create the necessary GraphQL type for ProductBlock and ensure proper registration with `register_graphql()`. This will help you avoid any infinite loop scenarios and enable smooth integration of domain models with GraphQL.
```